staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/device.go

```diff
@@ -268,25 +270,35 @@ func setVisitor(visitorConfig *driver.VisitorConfig, twin *common.Twin, dev *dri
 }
 
 // DevInit initialize the device
-func (d *DevPanel) DevInit(cfg *config.Config) error {
-	devs := make(map[string]*common.DeviceInstance)
-
-	switch cfg.DevInit.Mode {
-	//case common.DevInitModeConfigmap:
-	//	if err := parse.Parse(cfg.DevInit.Configmap, devs, d.models, d.protocols); err != nil {
-	//		return err
-	//	}
-	case common.DevInitModeRegister:
-		if err := parse.ParseByUsingRegister(devs, d.models, d.protocols); err != nil {
+func (d *DevPanel) DevInit(deviceList []*dmiapi.Device, deviceModelList []*dmiapi.DeviceModel) error {
+	if len(deviceList) == 0 || len(deviceModelList) == 0 {
+		return ErrEmptyData
+	}
+
+	for _, model := range deviceModelList {
+		cur := parse.ParseDeviceModelFromGrpc(model)
+		d.models[model.Name] = cur
+	}
+
+	for _, device := range deviceList {
+		commonModel := d.models[device.Spec.DeviceModelReference]
+		protocol, err := parse.BuildProtocolFromGrpc(device)
+		if err != nil {
 			return err
```

Comment:
Is there a clear enough error message here? If not clear enough, you can supplement the information